compiler/src/dmd/expressionsem.d

```diff
@@ -5897,6 +5897,19 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             printf("CallExp::semantic() %s\n", exp.toChars());
         }
 
+        scope (exit)
+        {
+            if (TypeFunction tf = exp.f ? cast(TypeFunction)exp.f.type : null)
+            {
+                result.rvalue = tf.isRvalue;
+                if (tf.isRvalue && !tf.isRef)
+                {
+                    error(exp.f.loc, "__rvalue only valid on functions that return ref");
```

Comment:
backticks around `__rvalue`

also "functions that return by `ref`" (not the be confused with `return ref`)